### PR TITLE
Topic feature flag env var

### PR DESCRIPTION
**Pull request by igorT**

### CODE REVIEW

File: packages/canary-features/addon/index.js [hunk 1]
```diff
@@ -1 +1,32 @@
...
+    )
+      continue;
+    let defaultValue = ENV[flag];
+    if (defaultValue === true) {
+      ENV[flag] = EmberENV[flag] !== false;
+    } else if (defaultValue === false) {
+      ENV[flag] = EmberENV[flag] === true;
+    }
+  }
+})(global.EmberENV || global.ENV);
```
Comment (rwjblue) [inline packages/canary-features/addon/index.js]: We should use EmberDataENV or something (not the same global that Ember uses).

File: packages/canary-features/addon/environment/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// export * from './lib/context';
```
Comment (rwjblue) [inline packages/canary-features/addon/environment/index.js]: Probably can kill these commented things

File: packages/canary-features/addon/index.js [hunk 3]
```diff
@@ -1 +1,32 @@
-// export const sample_feature_flag = true;
+import { assign } from '@ember/polyfills';
+import global from '@ember-data/canary-features/environment';
+// import { ENV } from '@ember-data/canary-features/
```
Comment (rwjblue) [inline packages/canary-features/addon/index.js]: Can cut this

File: packages/canary-features/addon/index.js [hunk 4]
```diff
@@ -1 +1,32 @@
-// export const sample_feature_flag = true;
```
Comment (rwjblue) [inline packages/canary-features/addon/index.js]: Same here

File: packages/canary-features/addon/index.js [hunk 5]
```diff
@@ -1 +1,32 @@
...
+    if (defaultValue === true) {
+      ENV[flag] = EmberENV[flag] !== false;
+    } else if (defaultValue === false) {
+      ENV[flag] = EmberENV[flag] === true;
+    }
+  }
+})(global.EmberENV || global.ENV);
+
+export const DEFAULT_FEATURES = {
+  sample_feature_flag: null,
```
Comment (rwjblue) [inline packages/canary-features/addon/index.js]: I _vaguely_ prefer the SCREAMING_SNAKE_CASE here, though I know this isn’t really new (or the point of the PR)

File: packages/canary-features/addon/index.js [hunk 6]
```diff
@@ -1 +1,32 @@
...
+};
+
+export const FEATURES = assign(DEFAULT_FEATURES, ENV.FEATURES);
```
Comment (rwjblue) [inline packages/canary-features/addon/index.js]: You don’t want to mutate DEFAULT_FEATURES here, probably want `assign({}, DEFAULT_FEATURES, ENV.FEATURES)`

File: packages/canary-features/addon/index.js [hunk 7]
```diff
@@ -1 +1,32 @@
...
+export const sample_feature_flag = DEFAULT_FEATURES.sample_feature_flag;
```
Comment (rwjblue) [inline packages/canary-features/addon/index.js]: Should be `= FEATURES.sample_feature_flag`

File: packages/canary-features/addon/index.js (hunk 1 above)
Comment (efx) [reply packages/canary-features/addon/index.js]: I'll get this and the remaining feedback addressed tomorrow.

File: packages/canary-features/addon/index.js (hunk 5 above)
Comment (efx) [reply packages/canary-features/addon/index.js]: Given Ember uses that same convention, I agree to make it the same. I don't see warrant for introducing a new convention.

**Comment by efx** [comment]
Addressed most of the feedback, but this still does not work when testing enabling a flag within an ember data file. I tested with the following in internal model's constructor:
```js
import { SAMPLE_FEATURE_FLAG } from '@ember-data/canary-features';
// ...
 if (SAMPLE_FEATURE_FLAG) {
   console.log('hi')
}                                                                       

I think I'm missing how exactly debug macros correctly serializes the environment / variables from features.js into a module the browser can understand.

**Comment by igorT** [comment]
@efx I tested it and it actually worked for me

**Comment by rwjblue** [comment]
FYI - Igor and I did some more work on the feature flagging setup over in #6197. AFAICT the main thing left is parsing out the default features and values from `@ember-data/canary-features`'s `addon/index.js` instead of hard coding the list in both places, there is a pending TODO in the file for that though...

**Pull request merged**